### DISCUSSION TITLE
Closing the Drawer component when used as a nav menu

### DISCUSSION THREAD

**Discussion by nerdstep** [palantir/blueprint#6545] [Q&A] [answered]
I'm using the `Drawer` component to render a nav menu for small viewports but I'm having a hard time figuring out the best way to close the Drawer component when menu item link is clicked. Something like the `POPOVER_DISMISS` class seems like it would fit my use-case but that doesn't seem to work here. Anyone have any other ideas or suggestions?

**Comment by adidahiya** [comment, marked as answer]
`Drawer` is a controlled component, so you need to set its `isOpen` state explicitly with that prop. There's no uncontrolled functionality that would allow something like the `Classes.POPOVER_DISMISS` class to work. So you will need to hook up your `MenuItem` click handler to set some state in your application which sends `isOpen={false}` to the relevant `Drawer`.

**Reply by nerdstep** [reply]
Thanks, that does makes sense. 😄 

Part of the problem is that I'm providing this as a higher level component in a library and will not have control over the child link elements. I was hoping to avoid using React context but that seems like the most appropriate solution in this scenario.
